test/getting-started/code/microvium/microvium.c

```diff
@@ -4761,9 +4762,22 @@ static TeError getProperty(VM* vm, Value objectValue, Value vPropertyName, Value
           }
         }
       }
-
-      *vPropertyValue = VM_VALUE_UNDEFINED;
-      return MVM_E_SUCCESS;
+      CODE_COVERAGE(326); // Hit
+      if (vPropertyName == VM_VALUE_STR_PROTO) {
+        CODE_COVERAGE(327); // Hit
+        *vPropertyValue = VM_VALUE_NULL;
+        return MVM_E_SUCCESS;
+      }
+      Value proto;
+      TeError e = getProperty(vm, objectValue, VM_VALUE_STR_PROTO, &proto);
+      if (e != MVM_E_SUCCESS) return e;
+      if (proto == VM_VALUE_NULL) {
+        CODE_COVERAGE(600); // Hit
+        *vPropertyValue = VM_VALUE_UNDEFINED;
+        return MVM_E_SUCCESS;
+      }
+      CODE_COVERAGE(601); // Hit
+      return getProperty(vm, proto, vPropertyName, vPropertyValue);
```

Comment:
This implementation recurses on the stack. The original implementation did the iteration within the single `while` loop, for stack efficiency.

Reply:
AFAIK this does _not_ recurse on the stack, due to tail call optimization (we are, in a sense, JMPing to getProperty).

Reply:
Ok, yes, it is a tail call, and probably the most common desktop compilers will perform tail call optimization. However, Microvium is optimized for 16-bit embedded platforms, and I'm not sure I can say the same for those. It doesn't seem like tail call optimization is guaranteed as part of the C/C++ spec, as far as I can tell. It's probably safer to just loop.

Reply:
It is 1000% safer to loop, especially on an arduino as... 8-bit AVR does *not* handle tail call optimizations apparantly (see https://github.com/wasm3/wasm3/blob/master/docs/Hardware.md, the bottommost target). I should change that and use dpProto.